internal/commands/parameters.go

```diff
@@ -101,22 +108,25 @@ func mergeBundleParameters(installation *store.Installation, ops ...mergeBundleO
 			return err
 		}
 	}
-	if err := matchAndMergeParametersDefinition(installation.Parameters, cfg.params, bndl.Definitions, cfg.strictMode, cfg.stderr); err != nil {
+	if err := matchAndMergeParametersDefinition(installation.Parameters, cfg); err != nil {
 		return err
 	}
 	var err error
-	installation.Parameters, err = bundle.ValuesOrDefaults(installation.Parameters, bndl)
+	installation.Parameters, err = bundle.ValuesOrDefaults(installation.Parameters, installation.Bundle)
 	return err
 }
 
-func matchAndMergeParametersDefinition(currentValues map[string]interface{}, parameterValues map[string]string, parameterDefinitions definition.Definitions, strictMode bool, stderr io.Writer) error {
-	for k, v := range parameterValues {
-		definition, ok := parameterDefinitions[k]
+func matchAndMergeParametersDefinition(currentValues map[string]interface{}, cfg *mergeBundleConfig) error {
+	for k, v := range cfg.params {
+		if cfg.verifyImmutable && cfg.bundle.Parameters.Fields[k].Immutable {
```

Comment:
Don't you think it should be the cnab runtime duty to check an immutable parameter is not set? So we would just let him fail if it happens?

Reply:
It definitely makes sense. It also means that we need to pass to the runtime which variables have been overridden.

Reply:
Fixed in cnab-go https://github.com/deislabs/cnab-go/pull/64